community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelStatement.java

```diff
@@ -130,23 +151,22 @@ void assertOpen()
 
     final void acquire()
     {
-        referenceCount++;
+        if ( referenceCount++ == 0 )
+        {
+            storeStatement.acquire();
+        }
     }
 
     private boolean release()
     {
-        referenceCount--;
-
-        return (referenceCount == 0);
+        return (--referenceCount == 0);
     }
```

Comment:
The `acquire` and `release` methods are not symmetric with regards to their visibility, or what they do to the `storeStatement`.